ewah/dag_factories/dag_factory_atomic.py

```diff
@@ -58,12 +58,8 @@ def raise_exception(msg: str) -> None:
         catchup = False
         assert croniter.is_valid(
             schedule_interval
-        ), "schedule_interval is neither timedelta nor not valid cron!"
-    else:
-        assert isinstance(
-            schedule_interval, timedelta
-        ), "schedule_interval must be cron-string or timedelta!"
-
+        ), "schedule_interval is not valid a cron expression!"
+    elif isinstance(schedule_interval, timedelta):
```

Comment:
Question (blocking): Are you sure this works as expected? After all, some variables like `catchup` look like they are not properly defined.

Reply:
Same as the other comment, I adjusted it to initialize the parameters before the if statements.